crates/uv/src/commands/tool/run.rs

```diff
@@ -118,9 +120,54 @@ pub(crate) async fn run(
         command.to_string_lossy(),
         args.iter().map(|arg| arg.to_string_lossy()).join(" ")
     );
-    let mut handle = process
-        .spawn()
-        .with_context(|| format!("Failed to spawn: `{}`", command.to_string_lossy()))?;
+    let mut handle = match process.spawn() {
+        Ok(handle) => Ok(handle),
+        Err(e) => {
+            let site_packages = SitePackages::from_environment(&environment)
+                .context("Failed to read site packages")?;
+            let package = PackageName::from_str(&from).context("Invalid package name {from}")?;
+
+            let installed = site_packages.get_packages(&package);
+            let Some(installed_dist) = installed.first().copied() else {
+                bail!("Expected at least one requirement")
+            };
+
+            let entry_points = entrypoint_paths(
+                &environment,
+                installed_dist.name(),
+                installed_dist.version(),
+            )
+            .context("Failed to read entrypoints")?;
+
+            if !entry_points
+                .iter()
+                .map(|e| e.0.as_str())
+                .any(|e| *e == *command)
+            {
+                writeln!(
+                    printer.stdout(),
+                    "The executable {} was not found.",
+                    command.user_display().red()
+                )?;
+                if !entry_points.is_empty() {
+                    writeln!(
+                        printer.stdout(),
+                        "However, the following executables are available: via `uv tool run --from {from} <EXECUTABLE>`",
+                    )?;
+                }
```

Comment:
Could it be ignored when user already used `--from`? I think he don't need to know about this hint. So IMO list of available executables is enough for him.
```rust
                if !entry_points.is_empty() {
                    let from_option_hint = if from == None { " via `uv tool run --from {from} <EXECUTABLE>`" } else { "" }
                    writeln!(
                        printer.stdout(),
                        "However, the following executables are available{from_option_hint}:",
                    )?;
                }
```